filebeat/module/kibana/log/ingest/pipeline.json

```diff
@@ -53,6 +53,14 @@
                 "ignore_missing": true
             }
         },
+        {
+            "script": {
+                "lang": "painless",
+                "source": "ctx.event.duration = Math.round(ctx.kibana.log.meta.res.responseTime * params.scale)",
+                "params": { "scale": 1000000 },
+                "if": "ctx.kibana.log.containsKey('meta') && ctx.kibana.log.meta.containsKey('res') && ctx.kibana.log.meta.res.containsKey('responseTime')"
```

Comment:
Nit: You could probably shorten this to something like: `ctx.kibana.log.meta?.res?.responseTime != null`.

Reply:
Oh man, Painless has that?  Thanks for pointing out ❤️

Reply:
Yep, it was added at some point: https://www.elastic.co/guide/en/elasticsearch/painless/current/painless-operators-reference.html#null-safe-operator

Reply:
@ycombinator Do you know which version this was added? Asking for compatibility reasons.

Reply:
5.1.1: https://github.com/elastic/elasticsearch/pull/21239

Reply:
Ah, then we are good :-)